src/System.Data.SqlClient/src/System/Data/SqlClient/SqlBuffer.cs

```diff
@@ -1027,90 +1027,47 @@ internal void SetToString(string value)
             _isNull = false;
         }
 
-        internal void SetToDate(byte[] bytes)
+        internal void SetToDate(ReadOnlySpan<byte> bytes)
         {
             Debug.Assert(IsEmpty, "setting value a second time?");
 
             _type = StorageType.Date;
-            _value._int32 = GetDateFromByteArray(bytes, 0);
+            _value._int32 = GetDateFromByteArray(bytes);
             _isNull = false;
         }
 
-        internal void SetToDate(DateTime date)
+        internal void SetToTime(ReadOnlySpan<byte> bytes, byte scale)
         {
             Debug.Assert(IsEmpty, "setting value a second time?");
-
-            _type = StorageType.Date;
-            _value._int32 = date.Subtract(DateTime.MinValue).Days;
-            _isNull = false;
-        }
-
-        internal void SetToTime(byte[] bytes, int length, byte scale)
-        {
-            Debug.Assert(IsEmpty, "setting value a second time?");
-
             _type = StorageType.Time;
-            FillInTimeInfo(ref _value._timeInfo, bytes, length, scale);
+            FillInTimeInfo(ref _value._timeInfo, bytes, scale);
             _isNull = false;
         }
 
-        internal void SetToTime(TimeSpan timeSpan, byte scale)
+        internal void SetToDateTime2(ReadOnlySpan<byte> bytes, byte scale)
```

Comment:
What does the `2` mean in this method name?  It's not very descriptive.

Reply:
It means that it's using the data layout for an sql type SQLDATETIME2, which represents a [datetime2](https://docs.microsoft.com/en-us/sql/t-sql/data-types/datetime2-transact-sql?view=sql-server-2017) and I agree that it's a silly name but that's what it's called.

Reply:
Ugh. Ok, thanks.